wp-content/themes/pub/wporg-learn-2020/archive-course.php

```diff
@@ -9,62 +9,101 @@
 
 namespace WordPressdotorg\Theme;
 
-$prev_category = '';
-$category_title = '';
-$category_description = '';
+$all_courses        = array();
+$course_categories  = array();
+$prev_category      = '';
+$begin_new_category = false;
 
 get_header();
 get_template_part( 'template-parts/component', 'breadcrumbs' );
 ?>
 
 <main id="main" class="site-main">
-
 	<section>
-		<div class="row align-middle between section-heading section-heading--with-space gutters">
-			<?php the_archive_title( '<h1 class="section-heading_title h2 col-8">', '</h1>' ); ?>
-			<?php get_template_part( 'template-parts/component', 'archive-search' ); ?>
+		<div class="section-heading section-heading--with-space">
+			<?php the_archive_title( '<h1 class="section-heading_title h2">', '</h1>' ); ?>
 		</div>
 
 		<hr>
 
-			<?php if ( have_posts() ) : ?>
-				<?php while ( have_posts() ) :
-					the_post();
+		<?php if ( have_posts() ) : ?>
+			<?php while ( have_posts() ) :
+				the_post();
 
-					$prev_category;
-					$begin_new_category = false;
-					$categories         = get_the_terms( get_the_ID(), 'course-category' );
+				$categories = get_the_terms( get_the_ID(), 'course-category' );
 
-					if ( isset( $categories[0] ) ) {
-						$category_title       = $categories[0]->name;
-						$category_description = $categories[0]->description;
-						$begin_new_category   = $category_title !== $prev_category;
-					}
+				if ( isset( $categories[0] ) ) {
+					$category_slug        = $categories[0]->slug;
+					$category_title       = $categories[0]->name;
+					$category_description = $categories[0]->description;
+
+					ob_start();
+					get_template_part(
+						'template-parts/component',
+						'card',
+						wporg_learn_get_card_template_args( get_the_ID() )
+					);
+
+					$all_courses[ $category_slug ]['name']        = $category_title;
+					$all_courses[ $category_slug ]['description'] = $category_description;
+					$all_courses[ $category_slug ]['courses'][]   = ob_get_clean();
+				}
+
+			endwhile;
+
+			if ( ! empty( $all_courses ) ) { ?>
+				<div class="section-intro">
+					<div class="row between gutters">
+						<p class="col-8"><?php esc_html_e( 'WordPress provides limitless ways for people to craft and grow their online presence. The content in these courses is delivered in multiple formats, with a focus on text and video, working towards practical learning objectives to help you become a better WordPress developer, designer, user, and contributor.', 'wporg-learn' ); ?></p>
+						<?php get_template_part( 'template-parts/component', 'archive-search' ); ?>
+					</div>
+				</div>
+				<nav class="section-nav">
+					<ul>
+					<?php foreach ( $all_courses as $slug => $category ) { ?>
+						<li class="section-nav-item">
+							<a href="#<?php echo esc_attr( $slug ); ?>"><?php echo esc_html( $category['name'] ); ?></a>
+						</li>
+					<?php } ?>
+					</ul>
+				</nav>
```

Comment:
Standard markup for an internal navigation, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav

Using semantic HTML for these helps assistive technologies.